src/AppBundle/Helper/ApiHelper.php

```diff
@@ -27,7 +31,18 @@ private function setUp($project)
     {
         if (!isset($this->api)) {
             $projectInfo = $this->labsHelper->databasePrepare($project);
-            $this->api = MediawikiApi::newFromPage($projectInfo['url']);
+
+            try {
+                if (isset($projectInfo["url"]) && strpos($projectInfo["url"], "api.php") !== false) {
```

Comment:
When is $projectInfo["url"] not set? Either this `isset` can go, or it should also be added to the 'else' clause below (to guard against trying to connect to the API on an unknown URL).